### ISSUE TITLE
Unable to modify roles via dashboard or API

### ISSUE THREAD

**Issue by Vicg853** [authorizerdev/authorizer#164]
**Version:** 0.22.0

**Describe the bug**
Errors with "invalid list of roles" although the inputted list is valid.

**Steps To Reproduce**
Simply by entering the dashboard and trying to append any user a role. Same behavior via API

- Update role mut
![image](https://user-images.githubusercontent.com/37887254/168135626-48ddadf2-9dcf-4fbf-a1c0-26d37776160a.png)
- Roles list query
![image](https://user-images.githubusercontent.com/37887254/168135888-5d8f8cc0-4cb4-49eb-be1e-72bec4a0258f.png)

**Desktop (please complete the following information):**

- OS: Windows (with WSL)
- Browser chrome  v101
- Version 0.22.0

**Comment by Vicg853** [comment]
Sorry for introducing a new issue inside another one. But they are mostly related.

I don't know if this is on purpose, but: shouldn't we be able to remove all roles from a user?

Currently, you can't do it, as it ignores passing through the update roles [handling](https://github.com/authorizerdev/authorizer/blob/773213e5a45ec51c07342743c946c26790d3e65e/server/resolvers/update_user.go#L129) when the role list is empty. 

I tried doing the following to fix it: ([here](https://github.com/authorizerdev/authorizer/blob/773213e5a45ec51c07342743c946c26790d3e65e/server/resolvers/update_user.go#L147))
```go
if rolesToSave != "" || len(params.Roles) == 0 {
      user.Roles = rolesToSave
}
```
And it worked!
But I never used Go, so maybe there is a better solution or practice to follow.

**Comment by lakhansamani** [comment]
@Vicg853 we don't want to assign empty roles to the user because it can mess up the authorization system, where roles are crucial. While signing up if no roles are passed `DEFAULT_ROLE` is assigned to the user.

**Comment by lakhansamani** [comment]
@Vicg853 I was able to update roles from dashboard.
I also tested them after restarting server, they were persisted.

Can you share how are setting up ROLES env, is it via `.env` file or via `Dashboard` only?
JFYI: `.env` file variables are given higher preference

<img width="1440" alt="Screenshot 2022-05-13 at 7 57 17 AM" src="https://user-images.githubusercontent.com/6964334/168199312-26566bf9-bb5e-45ae-aa34-62e446845edf.png">

Also tested the JWT token generated after updating role

<img width="1391" alt="Screenshot 2022-05-13 at 8 00 03 AM" src="https://user-images.githubusercontent.com/6964334/168199574-af9f4077-1fe9-4329-a2c5-61520997a5cc.png">

**Comment by lakhansamani** [comment]
Also tested above scenario via APIs, seems to be working fine, specially after merging ur PR :-) 

<img width="1436" alt="Screenshot 2022-05-13 at 8 05 45 AM" src="https://user-images.githubusercontent.com/6964334/168200834-99603672-5dbc-4197-a82b-f45f9d02405c.png">

<img width="1440" alt="Screenshot 2022-05-13 at 8 11 05 AM" src="https://user-images.githubusercontent.com/6964334/168200791-ab02a05e-e63c-4fd0-9695-a1d8a3f43861.png">

**Comment by Vicg853** [comment]
> @Vicg853 we don't want to assign empty roles to the user because it can mess up the authorization system, where roles are crucial. While signing up if no roles are passed `DEFAULT_ROLE` is assigned to the user.

Oh ok. Makes sense! 

Thanks for the explanation!

**Comment by Vicg853** [comment]
> @Vicg853 I was able to update roles from dashboard. I also tested them after restarting server, they were persisted.
> 
> Can you share how are setting up ROLES env, is it via `.env` file or via `Dashboard` only? JFYI: `.env` file variables are given higher preference
> ....
> Also tested the JWT token generated after updating role
> .....

I'm setting up my roles through the 3 env vars (PROTECTED_ROLES, ROLES,....). 

I modified them on the dashboard at some point, but only for testing and to double-check if it wasn't one of my typing mistakes 🤣 that was causing the issue.  

But I had no trouble modding the env roles, only the user's roles.

**Comment by Vicg853** [comment]
> Also tested above scenario via APIs, seems to be working fine, specially after merging ur PR :-)
> 

I'm going to check out the new patch and let you know.

**Comment by Vicg853** [comment]
@lakhansamani works fine!
Thanks for accepting the changes.

**Comment by lakhansamani** [comment]
Feel free to close this issue once you have verified the patch release

**Comment by Vicg853** [comment]
I think I spotted a side effect from what I pushed...

When using the login mutation, the user now must strictly have the same ``roles`` passed as params (e.g.: user ["a", "b"] and mut ["a"] returns invalid roles) 

The issue is probably [here](https://github.com/authorizerdev/authorizer/blob/ab717d956a47805d5a433982aaf48c4f676be12c/server/resolvers/login.go#L61). As I inverted the ``IsValidRoles`` params, now this line should be causing the behavior now seen with the mut.

Sorry for that. I'm going to pr a fix for it...
